text/0026-rest-api/rest-api.yaml

```diff
@@ -0,0 +1,1275 @@
+  /messages/by-tag/{tag}:
+    get:
+      tags:
+        - messages
+      summary: Find all messages that are marked by a given tag.
+      description: Find all messages that are marked by a given tag.
+      operationId: get-messages-by-tag
+      parameters:
+        - name: tag
+          in: path
+          description: ASCII encoded tag of the message.
+          required: true
+          schema:
+            type: string
+      responses:
+        "200":
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/GetMessagesByTagResponse'
+              examples:
+                Example 1:
+                  $ref: '#/components/examples/get-messages-by-tag-response-example-1'
+        "400":
+          description: unsuccessful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+              examples:
+                default:
+                  $ref: '#/components/examples/invalid-data-provided-response-example'
+        "404":
+          description: unsuccessful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+              examples:
+                default:
+                  $ref: '#/components/examples/can-not-find-data-response-example'
+  /messages/by-tag:
+    post:
+      tags:
+        - messages
+      summary: The bulk version of GET /messages/by-tag/{tag}.
+      description: The bulk version of GET /messages/by-tag/{tag}.
+      operationId: post-messages-by-tag
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/PostMessagesByTagParameters"
+        required: true
+      responses:
+        "200":
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/PostMessagesByTagResponse"
+              examples:
+                default:
+                  $ref: '#/components/examples/post-messages-by-tag-response-example-1'
+        "400":
+          description: unsuccessful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+              examples:
+                default:
+                  $ref: '#/components/examples/invalid-data-provided-response-example'
+        "404":
+          description: unsuccessful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+              examples:
+                default:
+                  $ref: '#/components/examples/can-not-find-data-response-example'
+  /messages/by-hash/{hash}/-is-referenced-by-milestone:
```

Comment:
What do you think about adding reference state in the response context of the messages by default? `/messages/by-hash/{hash}/-is-referenced-by-milestone` and `/transaction-messages/is-confirmed` can be removed in this way. And with reference information being included in the message object, it gives more clear view about the status of the message in the Tangle imho.

Reply:
You're talking only about `GET /messages/by-hash/{hash}/-is-referenced-by-milestone` and  `POST/messages/by-hash/-is-referenced-by-milestone`. You're not talking about removing confirmation endpoints right?

Reply:
I was thinking about general message/tx-message response in each api call could include the reference state already. So calling confirmation endpoints are not necessary in this way. Users could use one call to get every information about corresponding message.